pandas/indexes/multi.py

```diff
@@ -656,9 +665,74 @@ def _has_complex_internals(self):
         return True
 
     @cache_readonly
+    def is_monotonic(self):
+
+        def level_values(level):
+            unique = self.levels[level]
+            labels = self.labels[level]
+            return algos.take_1d(unique.values, labels,
+                                 fill_value=unique._na_value)
```

Comment:
for most dtypes this not s problem as we fill with iNaT for example in i8; only for floats this could be an issue i suppose

do you have an example that fails?